app/models/form/account_batch.rb

```diff
@@ -61,23 +64,31 @@ def accounts
   end
 
   def approve!
-    users = accounts.includes(:user).map(&:user)
-
-    users.each { |user| authorize(user, :approve?) }
-         .each(&:approve!)
+    accounts.includes(:user).find_each do |account|
+      approve_account(account)
+    end
```

Comment:
I think the rationale behind having the `authorize` calls first was to make sure we don't have a partial application of the batch actions. In practice, I don't think we can have the `authorize` call work for a user and fail for another, though.

Reply:
Yeah that's what I said below, but here it's for approve/reject